### PR TITLE
Mrb devex 166 kustomize

### PR DESCRIPTION
**Pull request by mikowitz**
Implement kustomize for avro-schema-registry.

* Add two "base" kustomize directories: one for web deployments and one for db deployments
* Add overlays for avro-schema-registry for each
* Update `Tiltfile.base` to use kustomize instead of `templatize_config`

### Still to do

* Add base kustomization for redis
* Implement for other apps

### CODE REVIEW

File: avro-schema-registry/db/kustomization.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+namePrefix: schema-registry-
+
+commonLabels:
+  app: schema-registry-db
+
+resources:
+- ../../support/templates/base/db
+
+patchesStrategicMerge:
```
Comment (djcp) [inline avro-schema-registry/db/kustomization.yaml]: How do we overwrite, say, the postgres image name? By editing `avro-schema-registry/db/deployment.yaml`?

File: support/scripts/shared_functions [hunk 2]
```diff
@@ -35,3 +35,8 @@ def check_repo(repo_name):
     fail("Expected to find a local repository at %s, but none was found" % (repo_name))
   else:
     pass
+
+def ez_kustomize(dir):
+  k = kustomize(dir)
+  s = str(k)
+  k8s_yaml(blob(s.replace('__HOME__', os.environ['HOME'])))
```
Comment (djcp) [inline support/scripts/shared_functions]: Yeah, that's better than shelling out to sed

File: support/setup [hunk 3]
```diff
@@ -0,0 +1,4 @@
+load('../support/scripts/shared_functions', 'enforce_local_env')
```
Comment (djcp) [inline support/setup]: Where is `support/setup` used?

File: avro-schema-registry/db/kustomization.yaml (hunk 1 above)
Comment (mikowitz) [reply avro-schema-registry/db/kustomization.yaml]: exactly

File: support/setup (hunk 3 above)
Comment (mikowitz) [reply support/setup]: oh. whoops. i think this was an artifact from my last refactor. This functionality has ended up in `Tiltfile.common`. Forgot I left this around when I `git add .` Can be removed.

**Comment by mikowitz** [comment]
@djcp I believe so, especially as we adopt it more widely.

**Pull request merged**